src/api.tsx

```diff
@@ -3,11 +3,28 @@ import Match from './models/match'
 import Analysis from './models/analysis'
 import Schema from './models/schema'
 import UserInfo from './models/user-info'
+import idbKeyval from 'idb-keyval'
 
 import { hasValidJWT, getJWT } from './utils'
 
 const endpoint = 'https://api.pigmice.ga'
 
+interface req {
+  path: string
+  method: string
+  body?: string
+}
+
+const addRequestToIdb = async (request: req) => {
+  const currentRequests = (await idbKeyval.get('cachedRequests')) as
+    | req[]
+    | undefined
+  if (currentRequests === undefined) {
+    return idbKeyval.set('cachedRequests', [request])
+  }
+  return idbKeyval.set('cachedRequests', currentRequests.concat(request))
```

Comment:
Wouldn't `else` make it more clear?

Reply:
No

Reply:
AFAIK early returns are favored over if-else in most languages (including JS).

Reply:
Less cyclomatic complexity is almost always better if possible.

Reply:
Fine I'll take the early return argument. However, it does not introduce any more cyclomatic complexity. There is still a branch at the if statement either way with two possiblities.

Reply:
True.